tracing-subscriber/src/layer.rs

```diff
@@ -494,6 +494,12 @@ where
     }
 
     fn try_close(&self, id: span::Id) -> bool {
+        let subscriber = &self.inner as &dyn Subscriber;
+        let _guard = match subscriber.downcast_ref::<Registry>() {
+            Some(registry) => Some(registry.ref_guard(id.clone())),
+            None => None,
+        };
```

Comment:
TIOLI: I think this could be 
```rust
let _guard = subscriber.downcast_ref::<Registry>()
    .and_then(|registry| registry.ref_guard(id.clone());
```
which could be a little more concise

Reply:
done!